<a href="https://colab.research.google.com/github/rimosh58/TU_math_2022/blob/main/Chap04_Cm_01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import scipy as sp
from scipy import stats
import pandas as pd
from pandas import Series, DataFrame

import matplotlib.pyplot as plt
import matplotlib as mpl
import seaborn as sns
%matplotlib inline

%precision 3

'%.3f'

In [2]:
import requests
import zipfile
from io import StringIO
import io

zip_file_url = "http://archive.ics.uci.edu/ml/machine-learning-databases/00356/student.zip"

r = requests.get(zip_file_url, stream = True)
z = zipfile.ZipFile(io.BytesIO(r.content))
z.extractall()

student_data_math = pd.read_csv("student-mat.csv", sep = ";")

In [3]:
student_data_math = pd.read_csv("student-mat.csv", sep = ";")

student_data_por = pd.read_csv("student-por.csv", sep = ";")

student_data_merge = pd.merge(student_data_math
                              , student_data_por
                              , on = ["school", "sex", "age", "address", "famsize", "Pstatus", "Medu"
                                      , "Fedu", "Mjob", "Fjob", "reason", "nursery", "internet"]
                              , suffixes = ("_math", "_por"))

In [17]:
# 4-1-1
print("数学の欠席数平均：", student_data_merge.absences_math.mean())
print("ポルトガル語の欠席数平均：", student_data_merge.absences_por.mean())
t, p1 = stats.ttest_rel(student_data_merge.absences_math, student_data_merge.absences_por)
print("p値 = ", p1)

数学の欠席数平均： 5.319371727748691
ポルトガル語の欠席数平均： 3.6727748691099475
p値 =  2.3441656888384195e-06


p値が0.05未満(p<0.05, 5%未満)を有意とみなす。<br>
欠席数に関するp値は2.34e-06である。<br>
したがって，p値が0.05未満なので， 差があると言える。

In [18]:
# 4-1-2
print("数学の欠席数平均：", student_data_merge.studytime_math.mean())
print("ポルトガル語の欠席数平均：", student_data_merge.studytime_por.mean())
t, p2 = stats.ttest_rel(student_data_merge.studytime_math, student_data_merge.studytime_por)
print("p値 = ", p2)

数学の欠席数平均： 2.0340314136125652
ポルトガル語の欠席数平均： 2.0392670157068062
p値 =  0.5643842756976525


p値が0.05未満(p<0.05, 5%未満)を有意とみなす。<br>
勉強時間に関するp値は0.56である。
したがって，p値が0.05以上なので，差があるとは言えない。